### ISSUE TITLE
Diffs highlighting substrings are hard to see in typescript files

### ISSUE THREAD

**Issue by xt0rted** [desktop/desktop#8574]
### Describe the bug

When a diff highlights a portion of the changed line in a typescript file the text is much harder to see with the dark theme than the light theme.

### Version & OS

- Version 2.2.2-beta3
- Windows 10

### Steps to reproduce the behavior

1.Modify a typescript file by adding an annotation to a line
2. View the diff in GitHub Desktop

### Expected behavior

To be able to more easily read the highlighted text.

### Actual behavior

Highlighted text is extremely difficult to see compared to the the light theme or a plain full line diff.

### Screenshots

![image](https://user-images.githubusercontent.com/831974/68081555-c4087880-fddd-11e9-838b-24802fad7da7.png)

![image](https://user-images.githubusercontent.com/831974/68081563-d2569480-fddd-11e9-9455-34091b3940bb.png)

![image](https://user-images.githubusercontent.com/831974/68081578-2c575a00-fdde-11e9-8b67-acfe91db8975.png)

**Comment by outofambit** [comment]
Hi @xt0rted thanks for filing this issue. I did some inspecting and the highlighted sections of a `+` diff line in the dark theme are indeed [lacking in contrast to meet accessibility standards](https://webaim.org/resources/contrastchecker/?fcolor=D2D4D9&bcolor=428342). The contrast ratio is about **3.1:1**, and needs to be at least **4.5:1** for text of that size.

I'm going to mark this as an accessibility bug and look into prioritizing it for an upcoming release. Thanks again!

**Comment by ampinsk** [comment]
Hi thanks for the issue @xt0rted! Did some investigation: Looks like in other spots with substring highlights on dark mode we change the text color to white, as you can see in this screenshot:

![Screen Shot 2019-12-16 at 10 40 34 AM](https://user-images.githubusercontent.com/10404068/71018787-23d58d00-20c7-11ea-9a6e-956e10cc3045.png)

So looks like the problem here is that the color is staying purple and not updating to white.

**Comment by tierninho** [comment]
Here is how it looks it v2.3.1, OSX:

![Screen Shot 2020-03-16 at 1 40 42 PM](https://user-images.githubusercontent.com/14828183/76808416-d0d5a980-678b-11ea-8f44-39112eab816a.png)
![Screen Shot 2020-03-16 at 1 40 23 PM](https://user-images.githubusercontent.com/14828183/76808419-d206d680-678b-11ea-891c-6b43120a13d1.png)

**Comment by sergiou87** [comment]
FWIW I just tested this on version 2.6.2-beta2, and it looks good on macOS 11.1:
![image](https://user-images.githubusercontent.com/1083228/103625678-97994080-4f3b-11eb-8f88-a3399f7dc9fb.png)

And Windows 10:
![image](https://user-images.githubusercontent.com/1083228/103625800-bf88a400-4f3b-11eb-9820-9d65fc998397.png)

As you see, it seems to behave well now. Could you please check if it still happens to you @xt0rted?

**Comment by xt0rted** [comment]
@sergiou87 yea the diff looks better now in 2.6.2-beta2 on windows.

You probably want this in a separate issue but I'm also seeing that any line using [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) isn't colored correctly.

![image](https://user-images.githubusercontent.com/831974/103717043-7e6fae80-4f92-11eb-97d9-8cdf07d5e56e.png)

![image](https://user-images.githubusercontent.com/831974/103717176-d1e1fc80-4f92-11eb-9fe2-90067b87cc21.png)

I ended up opening #11336 for this issue.

**Comment by sergiou87** [comment]
Thank you @xt0rted !! Seems like that one is already tracked, and this bug is fixed now. Closing… 😌
